ui/apps/platform/src/Containers/NetworkGraph/hooks/useNetworkPolicySimulator.ts

```diff
@@ -65,21 +66,19 @@ const defaultResultState = {
     isLoading: true,
 } as NetworkPolicySimulator;
 
-function useNetworkPolicySimulator({ simulation, clusterId }: UseNetworkPolicySimulatorParams): {
+function useNetworkPolicySimulator({ simulation }: UseNetworkPolicySimulatorParams): {
     simulator: NetworkPolicySimulator;
     setNetworkPolicyModification: SetNetworkPolicyModification;
 } {
+    const scopeHierarchy = useScopeHierarchy();
     const [simulator, setSimulator] = useState<NetworkPolicySimulator>(defaultResultState);
 
-    useEffect(() => {
+    useDeepCompareEffect(() => {
         setNetworkPolicyModification({
             state: 'ACTIVE',
-            options: {
-                clusterId,
-                searchQuery: '',
-            },
+            options: { scopeHierarchy },
         });
-    }, [clusterId, simulation.isOn]);
+    }, [scopeHierarchy, simulation.isOn]);
```

Comment:
since `scopeHierarchy` is a nested object, won't this be expensive? do we need to compare the whole object or can we compare just the cluster id as before?

Reply:
True, as it stands right now the query only requires the `clusterId` so the entire scope is not needed.

There is a comment above though from @sachaudh that says `// @TODO: Add the network search query as a second argument`. It seems like it would be consistent if the list of existing network policies in the simulator also matched the user's scope. It looks like the BE can support this, but at the expense of us having to do a separate network call for each namespace the user has selected. (The GetNetworkPolicies service call expects a single string value for `namespace`, unlike other more generic endpoints that accept a general query string.)

I can revert this bit for now and make a follow up ticket so that the scope of this PR doesn't increase any more.